torch/csrc/jit/codegen/cuda/executor_utils.cpp

```diff
@@ -487,17 +488,22 @@ kir::ExpressionEvaluator bindKernelInputs(
       for (size_t dim = 0; dim < root_domain.size(); dim++) {
         const auto extent = root_domain[dim]->extent();
         const auto value = aten_tensor.sizes()[dim];
-        const auto prev_value = expr_eval.evaluate(extent);
-        if (prev_value.has_value()) {
-          TORCH_CHECK(
-              *prev_value == value,
-              "Attempting to bind ",
-              kir::toString(extent),
-              " to ",
-              value,
-              "but it's already set to ",
-              *prev_value);
-        } else {
+        bool should_bind = true;
+        if (check_consistency) {
```

Comment:
When would this be false?

Reply:
The same consistency is done in `NaiveIntegerMachine::validate`. So this check is only necessary when using a legacy evaluator which this PR didn't attempt to remove.